pkg/auth/gitlab/creator.go

```diff
@@ -20,5 +20,6 @@ func (t *Creator) New(config auth.Configurator) (auth.Provider, error) {
 		ClientSecret:       cfg.ClientSecret,
 		GitLabOAuthBaseURL: fmt.Sprintf("https://%s/oauth", cfg.GitlabHostWithOptionalPort),
 		RedirectURL:        strings.TrimSuffix(cfg.TerralistSchemeHostAndPort, "/") + "/v1/api/auth/redirect",
+		Groups:             cfg.Groups,
```

Comment:
Instead of passing the CSV value to the provider and then splitting it with each request, I'd recommend splitting it here, and only passing the slice of groups to the provider.